### PR TITLE
Backport PR #50544 on branch 1.5.x (CI fix typing with latest ipython)

### PR DESCRIPTION
**Pull request by meeseeksmachine**
Backport PR #50544: CI fix typing with latest ipython

### CODE REVIEW

**Comment by phofl** [comment]
cc @MarcoGorelli Seems to be another typing failure

File: pandas/io/formats/printing.py [hunk 1]
```diff
@@ -259,9 +259,10 @@ def enable_data_resource_formatter(enable: bool) -> None:
         if mimetype not in formatters:
             # define tableschema formatter
             from IPython.core.formatters import BaseFormatter
+            from traitlets import ObjectName
 
             class TableSchemaFormatter(BaseFormatter):
-                print_method = "_repr_data_resource_"
+                print_method = ObjectName("_repr_data_resource_")
                 _return_type = (dict,)
```
Comment (MarcoGorelli) [inline pandas/io/formats/printing.py:266]:
```suggestion
                # expression has type "Tuple[Type[Dict[Any, Any]]]",
                # base class "BaseFormatter" defined the type as "Type[str]"
                _return_type = (dict,)  # type: ignore[assignment]
```

**Comment by MarcoGorelli** [comment]
the IPython and traitlets versions are the same on the 1.5.x branch, don't know why this is failing here but not on main, I can only put it down to it being a different mypy version. OK to `type: ignore`?

**Comment by phofl** [comment]
Ok by me

**Comment by mroeschke** [comment]
This might be failing on main https://github.com/pandas-dev/pandas/actions/runs/3838014340/jobs/6533977845

**Comment by MarcoGorelli** [comment]
yes, indeed - let's keep this to the first fix, and then backport https://github.com/pandas-dev/pandas/pull/50562 too?
/

**Pull request merged**